.github/workflows/test.yml

```diff
@@ -15,15 +15,27 @@ jobs:
         runs-on: macos-latest
         steps:
             - name: Checkout Branch
-              uses: actions/checkout@v3
+              uses: actions/checkout@v4
+
+            - name: Cache node modules
+              uses: actions/cache@v4
+              with:
+                # npm cache files are stored in `~/.npm` on Linux/macOS
+                path: ~/.npm
+                key: ${{ runner.os }}-node-modules-${{ hashFiles('**/package-lock.json') }}
+                restore-keys: ${{ runner.os }}-build-node-modules-
+
             - name: Setup Node
-              uses: actions/setup-node@v3
+              uses: actions/setup-node@v4
               with:
-                  node-version: 18.13.0
-            - name: Install
-              run: npm install
+                  node-version: '23.x'
```

Comment:
odd versions are the development versions, the stable versions are what they call "active LTS" or "maintenance LTS" on https://github.com/nodejs/release?tab=readme-ov-file#release-schedule

Let's use 22.x :-)

Reply:
Also it looks like it can use the value form `package.json` directly (see documentation in https://github.com/actions/setup-node) when using
```
 node-version-file: "package.json"
```

Might be worth trying :-)

Reply:
BTW you also need to update package.json of course :-)

and all comments also apply below